src/main.rs

```diff
@@ -73,6 +77,42 @@ macro_rules! unwrap_or_show_error {
     }};
 }
 
+fn get_mirror_url() -> Result<String> {
+    let mut status_file = File::open("/var/lib/apt/gen/status.json")?;
+    let mut status_data = String::new();
+    status_file.read_to_string(&mut status_data)?;
+
+    let status_v: jsonValue = serde_json::from_str(&status_data)?;
+    let mirror: String = match &status_v["mirror"][0] {
+        jsonValue::String(m) => String::from(m),
+        _ => {
+            return Err(format_err!("Malformed apt-gen-list data."));
+        }
+    };
+
+    let mut mirrors_file = File::open("/usr/share/distro-repository-data/mirrors.yml")?;
+    let mut mirrors_data = String::new();
+    mirrors_file.read_to_string(&mut mirrors_data)?;
+
+    let mirrors_v: yamlValue = serde_yaml::from_str(&mirrors_data)?;
+    let mirror_url = match &mirrors_v[&mirror]["url"] {
+        yamlValue::String(u) => String::from(u),
+        _ => {
+            return Err(format_err!("Malformed mirrors data."));
+        }
+    };
+
+    Ok(mirror_url)
+}
+
+lazy_static! {
+    static ref DEFAULT_MANIFEST_URL: String = format!(
+        "{}{}",
+        get_mirror_url().unwrap_or(String::from("https://repo.aosc.io/")),
+        "debs/manifest/topics.json"
+    );
+}
```

Comment:
Hmm, before digging into this, let me ask you a question: How many times are `DEFAULT_MANIFEST_URL` referenced?

Reply:
Once?

Reply:
> Once?

So in that case, do you really need a global reference?

Reply:
So I can just write `format!` in that scope?

Reply:
Correct.